README.md

```diff
@@ -11,18 +11,59 @@ A module to:
 1. install the [DataDog](http://www.datadoghq.com) agent
 2. to send reports of puppet runs to the Datadog service [Datadog](http://www.datadoghq.com/).
 
-Requirements
-------------
+Releases
+--------
 
-Puppet >=2.7.x and <=4.2.x (we may work with newer versions, but untested). For detailed informations on compatibility, check the [module page](https://forge.puppetlabs.com/datadog/datadog_agent) on the Puppet forge.
+There are currently two actively maintained versions for the puppet module. To users on puppet >= 4.6.x, and possibly some older 4.x puppets, we recommend you use version 2.0+ of the module. 
+
+For users running on older versions of puppet the legacy module, series 1.x, should support most use-cases. 
+
+Hopefully a majority of users will be able to use the newer module as many of the puppet versions supported in the 1.x series of the module have been EOL'd for a while now.
+
+Some features might be back-ported if there's enough demand for it, but in general only maintenance will be performed on the 1.x series of the module. Future feature development will be performed on the newer puppet module.
+
+
+Version 2.x
+-----------
 
-Installation
-------------
+## Requirements
+
+Puppet >=4.6.x and <=5.3.x (we may work with newer versions, but untested). For detailed informations on compatibility, check the [module page](https://forge.puppetlabs.com/datadog/datadog_agent) on the Puppet forge.
+
+## Installation
 
 Install `datadog_agent` as a module in your Puppet master's module path.
 
     puppet module install datadog-datadog_agent
 
+### Upgrade from the previous module version 1.x
+
+Most of the manifests should be backward compatible. However, there are important changes to the main manifest:
+
+- By default datadog agent 6.x will now be installed.
+  - You may still setup agent 5.x in your nodes by seting the `agent5_enable` boolean parameter to `True`
+- Enhanced support for our APM tracing agent
+- Enhanced support for our Process agent
+- Enhanced support for additional agent6 options
+- APM tracing and Process agents are now opt-in options and will be disabled by default.
+  - Main `datadog_agent` class has parameters to enable and configure the agents easily.
+- Increased granularity for apt/yum repo configuration 
+- New location for datadog puppet reporting configuration
+
+
+Version 1.x
+-----------
+
+## Requirements
+
+Puppet >=2.7.x and <=4.2.x (we may work with newer versions, but untested). For detailed informations on compatibility, check the [module page](https://forge.puppetlabs.com/datadog/datadog_agent) on the Puppet forge.
+
+## Installation
+
+Install `datadog_agent` as a module in your Puppet master's module path.
+
+    puppet module install datadog-datadog_agent --version 1.12.1
```

Comment:
Seems like we're stuck on 1.x for our puppet version. Should we be updating this module?